File: data structures and algorithms/cpp/06_121.cpp
std::vector<int> solution(int n, const std::vector<int>& a) {
    std::vector<int> padded(n + 2, 0);
    std::vector<int> result(n);
    
    // Copy array elements into padded array
    for (int i = 0; i < n; i++) {
        padded[i + 1] = a[i];
    }
    
    // Calculate sliding window sum
    for (int i = 0; i < n; i++) {
        result[i] = padded[i] + padded[i + 1] + padded[i + 2];
    }
    
    return result;
}

File: data structures and algorithms/cs/06_121.cs
public class Solution
{
   public static int[] Solution(int n, int[] a)
   {
      int[] result = new int[n];
      for (int i = 0; i < n; i++)
      {
         int leftValue = (i > 0) ? a[i - 1] : 0;
         int rightValue = (i < n - 1) ? a[i + 1] : 0;
         result[i] = leftValue + a[i] + rightValue;
      }
      return result;
   }
}
